internal/archive/archive_test.go

```diff
@@ -302,6 +302,38 @@ func testArchive(t *testing.T, when spec.G, it spec.S) {
 					verify.nextSymLink("/nested/dir/dir-in-archive/sub-dir/link-file", "../some-file.txt")
 				}
 			})
+
+			when.Focus("files are compressed in fat (MSDOS) format", func() {
+				it.Before(func() {
+					src = filepath.Join("testdata", "fat-zip-to-tar.zip")
+				})
+
+				it("writes a tar to the dest dir with 0777", func() {
+					fh, err := os.Create(filepath.Join(tmpDir, "some.tar"))
+					h.AssertNil(t, err)
+
+					tw := tar.NewWriter(fh)
+
+					err = archive.WriteZipToTar(tw, src, "/nested/dir/dir-in-archive", 1234, 2345, -1, true, nil)
+					h.AssertNil(t, err)
+					h.AssertNil(t, tw.Close())
+					h.AssertNil(t, fh.Close())
+
+					file, err := os.Open(filepath.Join(tmpDir, "some.tar"))
+					h.AssertNil(t, err)
+					defer file.Close()
+
+					tr := tar.NewReader(file)
+
+					verify := tarVerifier{t, tr, 1234, 2345}
+					// FAT stores names in uppercase: https://superuser.com/a/1297675
```

Comment:
Not sure I'm convinced of this comment based on the contents of the jar in the originating issue (see below). I don't know how the generated file was constructed and maybe the tool used created this artifact but we should remove this comment to prevent unintentionally misinforming people of the contents in all fat formatted zip files.

```
# ORIGINATING FILE
└─▪ zipinfo -m /Users/javier.romero/Downloads/webgoat-container-7.1-exec.jar | head -10
Archive:  /Users/javier.romero/Downloads/webgoat-container-7.1-exec.jar
Zip file size: 74202267 bytes, number of entries: 4239
-rw----     2.0 fat 64791363 bX  0% defN 16-Nov-18 16:51 WebGoat.war
-rw----     2.0 fat   152716 bl 89% defN 16-Nov-18 16:51 conf/web.xml
-rw----     2.0 fat      271 bl 21% defN 16-Nov-18 16:51 tomcat.standalone.properties
-rw----     2.0 fat      638 bl 40% defN 16-Nov-18 16:51 javax/annotation/Generated.class
-rw----     2.0 fat      453 bl 40% defN 16-Nov-18 16:51 javax/annotation/ManagedBean.class
-rw----     2.0 fat      396 bl 40% defN 16-Nov-18 16:51 javax/annotation/PostConstruct.class
-rw----     2.0 fat      390 bl 40% defN 16-Nov-18 16:51 javax/annotation/PreDestroy.class
-rw----     2.0 fat     1179 bl 51% defN 16-Nov-18 16:51 javax/annotation/Resource$AuthenticationType.class


# TEST FILE
└─▪ zipinfo -m ~/Downloads/fat-zip-to-tar.zip
Archive:  /Users/javier.romero/Downloads/fat-zip-to-tar.zip
Zip file size: 502 bytes, number of entries: 3
-rw----     2.0 fat       12 bx  0% stor 20-Apr-21 07:29 SOME-FIL.TXT
drwx---     2.0 fat        0 bx  0% stor 20-Apr-21 07:29 SUB-DIR/
-rw----     2.0 fat       12 bx  0% stor 20-Apr-21 07:29 SUB-DIR/LINK-FIL
3 files, 24 bytes uncompressed, 24 bytes compressed:  0.0%
```

Reply:
You're right, that seems to be something weird when using `zip` on a linux machine. I recreated the dir by zipping it on a windows VM, and it is lower-case